### PR TITLE
refactor: handle exceptions when updating addresses

### PR DESCRIPTION
**Pull request by scmmishra**
When creating a customer, it maybe the case that one of the addresses may not be present. This PR manages those conditions

### CODE REVIEW

**Comment by scmmishra** [comment]
@Mergifyio backport version-12-hotfix

File: erpnext/erpnext_integrations/connectors/woocommerce_connection.py [hunk 1]
```diff
@@ -73,10 +73,25 @@ def link_customer_and_address(raw_billing_data, raw_shipping_data, customer_name
...
+		
+		try:
+			shipping_address = frappe.get_doc("Address", {"woocommerce_email": customer_woo_com_email, "address_type": "Shipping"})
+			rename_address(shipping_address, customer)
+		except (
+			frappe.DoesNotExistError,
+			frappe.DuplicateEntryError,
+			frappe.ValidationError,
+		):
+			pass
```
Comment (marination) [inline erpnext/erpnext_integrations/connectors/woocommerce_connection.py:94]:
```suggestion
              address_types = ["Billing", "Shipping"]
		for address_type in address_types:
                      if frappe.db.exists("Address", {"woocommerce_email": customer_woo_com_email, "address_type": address_type}):
                            address = frappe.get_doc("Address", {"woocommerce_email": customer_woo_com_email, "address_type": address_type})
                            rename_address(address, customer)
```

Comment (marination) [inline erpnext/erpnext_integrations/connectors/woocommerce_connection.py]:
```py
address_types = ["Billing", "Shipping"]
      for address_type in address_types:
              if frappe.db.exists("Address", {"woocommerce_email": customer_woo_com_email, "address_type": address_type}):
                            address = frappe.get_doc("Address", {"woocommerce_email": customer_woo_com_email, "address_type": address_type})
                            rename_address(address, customer)
```

**Review by marination — APPROVED** [review]

**Comment by marination** [comment]
@Mergifyio backport v12-pre-release

**Pull request merged**